internal/host/validator.go

```diff
@@ -677,48 +679,47 @@ func (v *validator) hasSufficientNetworkLatencyRequirementForRole(c *validationC
 	if len(c.host.Connectivity) == 0 {
 		return ValidationPending
 	}
-	latency, err := v.validateNetworkLatencyForRole(c.host, c.clusterHostRequirements, c.cluster.Hosts)
-	if err != nil || len(latency) > 0 {
-		return ValidationFailure
-	}
-	return ValidationSuccess
+	status, _, _ := v.validateNetworkLatencyForRole(c.host, c.clusterHostRequirements, c.cluster.Hosts, c.cluster.OpenshiftVersion, c.cluster.EmailDomain)
+	return status
 }
 
-func (v *validator) validateNetworkLatencyForRole(host *models.Host, clusterRoleReqs *models.ClusterHostRequirements, hosts []*models.Host) ([]string, error) {
+func (v *validator) validateNetworkLatencyForRole(host *models.Host, clusterRoleReqs *models.ClusterHostRequirements, hosts []*models.Host, openshiftVersion, emailDomain string) (ValidationStatus, []string, error) {
 	var connectivityReport *models.ConnectivityReport
 	connectivityReport, err := hostutil.UnmarshalConnectivityReport(host.Connectivity)
 	if err != nil {
-		return nil, fmt.Errorf("unable to unmarshall host connectivity for %s:%s", host.ID, err)
+		return ValidationFailure, nil, fmt.Errorf("unable to unmarshall host connectivity for %s:%s", host.ID, err)
 	}
 	failedHostIPs := map[string]struct{}{}
 	failedHostNames := []string{}
 	for _, r := range connectivityReport.RemoteHosts {
 		for _, l3 := range r.L3Connectivity {
+			hostname, role, err := GetHostnameAndRoleByIP(l3.RemoteIPAddress, hosts)
+			if err != nil {
+				v.log.Warnf("unable to determine host's role and hostname for IP %s: %s", l3.RemoteIPAddress, err)
+				return ValidationFailure, nil, fmt.Errorf("unable to determine role and hostname for IP %s: %s", l3.RemoteIPAddress, err)
+			}
+			v.metricsAPI.NetworkLatencyBetweenHosts(openshiftVersion, emailDomain, host.Role, role, l3.AverageRTTMs)
 			if l3.AverageRTTMs > *clusterRoleReqs.Total.NetworkLatencyThresholdMs {
-				if _, ok := failedHostIPs[l3.RemoteIPAddress]; !ok {
-					hostname, role, err := GetHostnameAndRoleByIP(l3.RemoteIPAddress, hosts)
-					if err != nil {
-						v.log.Warnf("unable to determine host's role and hostname for IP: %s", err)
-						continue
-					}
-					if role == host.Role {
-						failedHostIPs[l3.RemoteIPAddress] = struct{}{}
-						failedHostNames = append(failedHostNames, hostname)
-					}
+				if _, ok := failedHostIPs[l3.RemoteIPAddress]; !ok && role == host.Role {
+					failedHostIPs[l3.RemoteIPAddress] = struct{}{}
+					failedHostNames = append(failedHostNames, hostname)
```

Comment:
What about sending metrics?

Reply:
It was not considered when I discussed this with @gamli75 and @avishayt . What  we are aiming to do is capture the latency in a histogram. Knowing the latency limits between roles, we can easily extract the percentage of installations where the limit was crossed for a given set of roles.